Framework/Geometry/src/Instrument/Goniometer.cpp

```diff
@@ -390,10 +394,19 @@ void Goniometer::loadNexus(::NeXus::File *file, const std::string &group) {
   file->readData("num_axes", num_axes);
   motors.clear();
   motors.reserve(num_axes);
-  for (int i = 0; i < num_axes; i++) {
-    GoniometerAxis newAxis;
-    newAxis.loadNexus(file, "axis" + Strings::toString(i));
-    motors.emplace_back(newAxis);
+  // If a rotation matrix is available and no rotation axes are provided load should behave like initFromR
+  if (file->hasData("rotation_matrix") && (num_axes < 1)) {
+    std::vector<double> matrixData;
+    file->readData("rotation_matrix", matrixData);
+    DblMatrix rotMat(matrixData);
+    setR(rotMat);
+    initFromR = true; // set as initFromR to prevent overwrite on R recalc
```

Comment:
Very minor -but I think the naming convention for such member variable (if that is the right term?) is `m_initFromR` - obviously this wasn't introduced in this PR (apologies for missing it before), but it helps to make the scope of the variable clear!